charms/knative-operator/src/charm.py

```diff
@@ -4,16 +4,24 @@
 
 """A Juju Charm for knative-operator."""
 
+import glob
 import logging
 import traceback
-
-from jinja2 import Environment, FileSystemLoader
-from lightkube import ApiError, Client, codecs
+from pathlib import Path
+
+from charmed_kubeflow_chisme.exceptions import ErrorWithStatus
+from charmed_kubeflow_chisme.kubernetes import (  # noqa N813
+    KubernetesResourceHandler as KRH,
+)
+from charmed_kubeflow_chisme.lightkube.batch import delete_many
+from lightkube import ApiError
 from ops.charm import CharmBase
 from ops.main import main
 from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus
 from ops.pebble import ChangeError, Layer
 
+REQUEST_LOG_TEMPLATE = '{"httpRequest": {"requestMethod": "{{.Request.Method}}", "requestUrl": "{{js .Request.RequestURI}}", "requestSize": "{{.Request.ContentLength}}", "status": {{.Response.Code}}, "responseSize": "{{.Response.Size}}", "userAgent": "{{js .Request.UserAgent}}", "remoteIp": "{{js .Request.RemoteAddr}}", "serverIp": "{{.Revision.PodIP}}", "referer": "{{js .Request.Referer}}", "latency": "{{.Response.Latency}}s", "protocol": "{{.Request.Proto}}"}, "traceId": "{{index .Request.Header "X-B3-Traceid"}}"}'
```

Comment:
Why separate this out?

Reply:
There are double {{ }} in the string so when we use jinja templating it throws an error :(. Could not find any better solution.